### PR TITLE
PLT-8104: @here should not be added to search on clicking @ icon to display mentions

### PR DESCRIPTION
**Pull request by MusikPolice**
#### Summary
Explicitly removed @here from search terms when the mentions button is pressed, regardless of notification preferences. Tidied up search term removal code to avoid unnecessary whitespace between search terms

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-8104

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)

### CODE REVIEW

File: actions/global_actions.jsx [hunk 1]
```diff
@@ -474,12 +474,19 @@ export function emitSearchMentionsEvent(user) {
...
-        if (termKeys.indexOf('@all') !== -1) {
-            termKeys[termKeys.indexOf('@all')] = '';
+        const indexOfAll = termKeys.indexOf('@all');
+        if (indexOfAll !== -1) {
+            termKeys.splice(indexOfAll, 1);
+        }
+
+        const indexOfHere = termKeys.indexOf('@here');
+        if (indexOfHere !== -1) {
+            termKeys.splice(indexOfHere, 1);
```
Comment (MusikPolice) [inline actions/global_actions.jsx]: This is the actual bug fix - suspect that we changed default mention keys to include `@here`, but didn't update this code. The rest of the change is just removing terms from the array, rather than replacing them with empty string to avoid building up unnecessary whitespace when the array elements are joined on line 492

Comment (saturninoabril) [reply actions/global_actions.jsx]: I've checked this issue and it might be related to this added key here - https://github.com/mattermost/mattermost-webapp/blob/master/stores/user_store.jsx#L464
